packages/beacon-node/src/chain/validation/dataColumnSidecar.ts

```diff
@@ -45,14 +47,60 @@ export async function validateGossipDataColumnSidecar(
 }
 
 export function validateDataColumnsSidecars(
-  _blockSlot: Slot,
-  _blockRoot: Root,
-  _expectedKzgCommitments: deneb.BlobKzgCommitments,
-  _dataColumnSidecars: electra.DataColumnSidecars,
-  _opts: {skipProofsCheck: boolean} = {skipProofsCheck: false}
+  blockSlot: Slot,
+  blockRoot: Root,
+  dataColumnSidecars: electra.DataColumnSidecars,
+  opts: {skipProofsCheck: boolean} = {skipProofsCheck: false}
 ): void {
-  // stubbed
-  return;
+  const commitmentBytes: Uint8Array[] = [];
+  const cellIndices: number[] = [];
+  const cells: Uint8Array[] = [];
+  const proofBytes: Uint8Array[] = [];
+  for (let sidecarsIndex = 0; sidecarsIndex < dataColumnSidecars.length; sidecarsIndex++) {
+    const columnSidecar = dataColumnSidecars[sidecarsIndex];
+    const columnIndex = columnSidecar.index;
+    const columnBlockHeader = columnSidecar.signedBlockHeader.message;
+    const columnBlockRoot = ssz.phase0.BeaconBlockHeader.hashTreeRoot(columnBlockHeader);
+    if (columnBlockHeader.slot !== blockSlot || !byteArrayEquals(columnBlockRoot, blockRoot)) {
+      throw new Error(
+        `Invalid columnSidecar with slot=${columnBlockHeader.slot} columnBlockRoot=${toHex(columnBlockRoot)} columnIndex=${columnIndex} for the block blockRoot=${toHex(blockRoot)} slot=${blockSlot} sidecarsIndex=${sidecarsIndex}`
+      );
+    }
+
+    if (columnIndex < NUMBER_OF_COLUMNS) {
+      throw new Error(
+        `Invalid data column columnIndex=${columnIndex} in slot=${blockSlot} blockRoot=${toHex(blockRoot)} sidecarsIndex=${sidecarsIndex}`
+      );
+    }
+
+    const {column, kzgCommitments, kzgProofs} = columnSidecar;
```

Comment:
you need to match these commitments with the expected commitments from the block

Reply:
good catch.  updated!